src/components/UserProfile/UserProfile.js

```diff
@@ -1,22 +1,54 @@
-import React, {useState, useEffect} from 'react'
+import React, {useState, useEffect, useContext } from 'react'
 import { useQuery, useMutation } from "@apollo/client";
 import FollowersList from './FollowersList/FollowersList'
 import Post from '../Post/Post.js'
+import UserContext from "../UserProfile/UserContext";
+import Loading from '../Loading/Loading.js'
 import * as gql from '../../queries/queries'
+
 import './UserProfile.css'
 
 const UserProfile = ({user}) => {
-  const { loading, error, data } = useQuery(gql.GET_USER_INFO(user));
+  const value = useContext(UserContext)
+  const GetFollowingInfo = useQuery(gql.GET_FOLLOWER_INFO(user));
+  const GetVisitedUserInfo = useQuery(gql.GET_USER_INFO(user));
+  const [clicked, setClicked] = useState(false)
   const [profile, setProfile] = useState('')
   const [followersVisible, setFollowersVisible] = useState(false)
+
+  const [followUser] = useMutation(gql.CREATE_FOLLOWER, {
+    refetchQueries: [{ query: gql.GET_FOLLOWING_INFO(value) }, { query: gql.GET_FOLLOWER_INFO(user) }, { query: gql.GET_USER_INFO(user) }],
+  });
+
+  const [unFollowUser] = useMutation(gql.DELETE_FOLLOWER, {
+    refetchQueries: [{ query: gql.GET_FOLLOWING_INFO(value) }, { query: gql.GET_FOLLOWER_INFO(user) }, { query: gql.GET_USER_INFO(user) }, { query: gql.GET_USER_INFO(value) }],
+  });
+
   useEffect(() => {
-    let mounted = true;
-    if (mounted && data) check()
-    return () => mounted = false;
-  })
+    setClicked(false)
+  }, [GetVisitedUserInfo.data])
+
+  if (GetFollowingInfo.loading) {
+    return <Loading loading={GetFollowingInfo.loading} />
```

Comment:
Again I appreciate the consistency of you calling the loading component everywhere it needs to be used.